llvm/include/llvm/FuzzMutate/IRMutator.h

```diff
@@ -145,6 +145,8 @@ class InsertFunctionStrategy : public IRMutationStrategy {
 
   using IRMutationStrategy::mutate;
   void mutate(BasicBlock &BB, RandomIRBuilder &IB) override;
+
+  virtual Function *chooseFunction(Module *M, RandomIRBuilder &IB);
```

Comment:
Good idea abstracting them away. Maybe you can make it `protected`. Also maybe add comments to the header.